### PR TITLE
Add new utility slash commands

### LINKED ISSUE #118: Slash commands to leverage new DB features
(How do mentions function in hidden messages?)

Slash commands can return input that's hidden from a channel thus solving the issue of spam.

/info or /about in carrier channel to show data on that carrier including @owner and @crew role without pinging
/find or /carrier to search for a specific carrier
/owner to search for carriers belonging to a user
/missions to list all missions without spamming channel (but with note about trade-alerts channel)

### PR DESCRIPTION
**Pull request by cimspin**
Closes #118 

New private slash commands:
`/mission`
`/missions`
`/info`
`/find <carriername>`
`/owner @owner`

### CODE REVIEW

File: MissionAlertBot.py [hunk 1]
```diff
@@ -314,6 +314,24 @@ def find_carrier_with_role_id(roleid):
...
+    Returns all carriers matching the ownerid
+
+    :param int ownerid: The owner id to match
+    :returns: A list of carrier data objects
+    :rtype: list[CarrierData]
+    """
+    carrier_db.execute(
+        f"SELECT * FROM carriers WHERE ownerid LIKE (?)", (f'%{ownerid}%',)
+    )
+    carrier_data = [CarrierData(carrier)  for carrier in carrier_db.fetchall() ]
```
Comment (gcruicks) [inline MissionAlertBot.py:328]: extra white space here before 'for'

Comment (gcruicks) [reply MissionAlertBot.py:328]: Empty whitespace at the end of the line that is not required also

File: MissionAlertBot.py [hunk 2]
```diff
@@ -314,6 +314,24 @@ def find_carrier_with_role_id(roleid):
...
+def find_carrier_with_owner_id(ownerid):
+    """
+    Returns all carriers matching the ownerid
+
+    :param int ownerid: The owner id to match
+    :returns: A list of carrier data objects
+    :rtype: list[CarrierData]
+    """
+    carrier_db.execute(
+        f"SELECT * FROM carriers WHERE ownerid LIKE (?)", (f'%{ownerid}%',)
```
Comment (gcruicks) [inline MissionAlertBot.py:326]: First one does not need to be an f-String, we don't insert anything

File: MissionAlertBot.py [hunk 3]
```diff
@@ -314,6 +314,24 @@ def find_carrier_with_role_id(roleid):
...
+    )
+    carrier_data = [CarrierData(carrier)  for carrier in carrier_db.fetchall() ]
+    for carrier in carrier_data:
+        print(f"FC {carrier.pid} is {carrier.carrier_long_name} {carrier.carrier_identifier} called by "
+              f"shortname {carrier.carrier_short_name} with channel <#{carrier.channel_id}> "
+              f"and owner {carrier.ownerid} and role id: {carrier.roleid} called from find_carrier_with_owner_id.")
```
Comment (gcruicks) [inline MissionAlertBot.py:332]: We should probably wrap this in a method, we do the same thing very often - not related to this, but would be nice to do.

File: MissionAlertBot.py [hunk 4]
```diff
@@ -957,6 +975,108 @@ def _mission_summary_embed(mission_data, embed):
...
+    channel = bot.get_channel(bot_spam_id)
+    await channel.send(f"{ctx.author} used /owner in <#{ctx.channel.id}> to find carriers owned by user with ID {stripped_owner}")
+    print(f"{ctx.author} used /owner in {ctx.channel} to find carriers owned by user with ID {stripped_owner}")
+
+    try:
+        owner = await bot.fetch_user(stripped_owner)
+        print(f"Found user as {owner.display_name}")
+    except HTTPException:
+        await ctx.send("Couldn't find any users by that name.", hidden=True)
+        raise EnvironmentError(f'Could not find Discord user matching ID {stripped_owner}')
```
Comment (gcruicks) [inline MissionAlertBot.py:997]: We should return the input value also here:

raise EnvironmentError(f'Could not find Discord user matching ID {at_owner_discord} ({stripped_owner})')

File: MissionAlertBot.py [hunk 5]
```diff
@@ -957,6 +975,108 @@ def _mission_summary_embed(mission_data, embed):
...
+
+        await ctx.send(embed=embed, hidden=True)
+
+    except TypeError as e:
+        print('Error: {}'.format(e))
+
+
+# mission slash command - private, non spammy
+@slash.slash(name="mission", guild_ids=[bot_guild_id],
+             description="Use in a Fleet Carrier's channel to privately display any active mission.")
```
Comment (gcruicks) [inline MissionAlertBot.py:1023]: A lot of this method looks to be a duplicate from the current mission command. Long term we need to think how to handle this, duplicate code is a pain to maintain

**Review by gcruicks — CHANGES_REQUESTED** [review]
Couple of minor things to change

File: MissionAlertBot.py (hunk 5 above)
Comment (cimspin) [reply MissionAlertBot.py:1023]: My thinking is more along the lines of removing the m.ission command but if it sticks around we can try to fold all the shared stuff into a single function

Comment (gcruicks) [reply MissionAlertBot.py:1023]: Open an issue to deprecate it then

**Review by gcruicks — APPROVED** [review]

**Pull request merged**